cmake/build_ace_tao.cmake

```diff
@@ -37,7 +37,9 @@ execute_process(
 # Get the C++ standard OpenDDS is going to be built with. We are going to force
 # the ACE/TAO build to use the same standard.
 set(_opendds_idl_std "$<TARGET_PROPERTY:opendds_idl,CXX_STANDARD>")
-set(_opendds_dcps_std "$<TARGET_PROPERTY:OpenDDS_Dcps,CXX_STANDARD>")
+if(NOT OPENDDS_JUST_BUILD_HOST_TOOLS)
+  set(_opendds_dcps_std "$<TARGET_PROPERTY:OpenDDS_Dcps,CXX_STANDARD>")
+endif()
 set(_opendds_std "$<IF:$<TARGET_EXISTS:OpenDDS_Dcps>,${_opendds_dcps_std},${_opendds_idl_std}>")
```

Comment:
So I was trying to remember if there was a reason why the "if OpenDDS_Dcps" decision is being made in a generator statement and I don't really know if there was a purpose. I also don't want to it rely on `OPENDDS_JUST_BUILD_HOST_TOOLS` though, because that's slightly indirect to what it's testing. I'm thinking something like:

```cmake
if(TARGET OpenDDS_Dcps)
  set(_opendds_std_target OpenDDS_Dcps)
else()
  set(_opendds_std_target opendds_idl)
endif()
set(_opendds_std "$<TARGET_PROPERTY:${_opendds_std_target},CXX_STANDARD>")
```

Reply:
Okay, so there was a purpose to it all being a generator statement and now I feel dumb. The targets don't exist before this, so what I wrote always results in it using `opendds_idl`, which won't work on the cross compiled build. It has to be generator statements or based on `OPENDDS_JUST_BUILD_HOST_TOOLS`.